src/verifast1.ml

```diff
@@ -3267,8 +3269,12 @@ module VerifyProgram1(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
         if ms = [] then on_fail () else
         let argtps = List.map (fun e -> let (_, tp, _) = (check e) in tp) args in
         (* Select the real methods with the correct number of type parameters *)
-        let ms = List.map (fun (sign, (tn', lm, gh, rt, xmap, pre, post, epost, terminates, fb', v, abstract)) ->
-          let targenv = if inAnnotation <> Some(true) then List.map2 (fun a b -> (a, b)) class_tparams class_targs else []

```

Comment:
Ik besef nu dat dit niet correct is voor statische methodes: de typeparameters van een klasse zijn niet in scope in statische methodes van de klasse. Dit is een fout in de vorige PR.

Reply:
Dit gaat op enkele andere plaatsen ook nog fout. Ik lost het op!